compiler/rustc_middle/src/ty/placeholder.rs

```diff
@@ -0,0 +1,18 @@
+use crate::ty::{BoundRegion, BoundTy, BoundVar, UniverseIndex};
+
+pub type PlaceholderRegion = Placeholder<BoundRegion>;
```

Comment:
Each of these should live in a module having to do with their specific kind I think